share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -398,7 +398,22 @@ function(ExternalZephyrProject_Add)
     # unless <image>_BOARD is defined.
     if(DEFINED ZBUILD_BOARD_REVISION)
       # Use provided board revision
-      set_target_properties(${ZBUILD_APPLICATION} PROPERTIES BOARD ${ZBUILD_BOARD}@${ZBUILD_BOARD_REVISION})
+      if(ZBUILD_BOARD MATCHES "/")
```

Comment:
this provides a narrow fix for the BOARD_REVISION, but question is if we should actually take a step back an do a proper cleanup of this ?

With HWMv2 we now have `BOARD_TARGET`, which is composed of `BOARD`, `BOARD_QUALIFIERS`, and `BOARD_REVISION`.
I think it would be good to adjust `ExternalZephyrProject_Add()` to allow: `BOARD_TARGET` or the individual parts where omitted parts will reuse the part from the board given to sysbuild.
That will also allow us to cleanup cases where inclusion of a image just want to adjust the qualifier and keep the rest as is.
Thus the signature can look as:
```
ExternalZephyrProject_Add(... [BOARD <board>] [BOARD_REVISION <revision>] [BOARD_QUALIFIER <qualifier>])
or
ExternalZephyrProject_Add(... [BOARD_TARGET <board-target>])
```

Current:
https://github.com/zephyrproject-rtos/zephyr/blob/dcce23c2e4fcfe08ca63861c2c4c0ff06e8bd250/share/sysbuild/cmake/modules/sysbuild_extensions.cmake#L204-L206

It's ok to go for the narrow fix for now, and then we can open an enhancement issue for the proposal in this comment.

Reply:
A post 4.0 cleanup of this is a good idea, which would also allow for dropping support for the legacy hwmv1 handling too